pylele_solid.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+
+"""
+    Abstract Base and Concrete Classes for all Gugulele Solid objects
+"""
+
+from __future__ import annotations
+
+import os
+import sys
+import argparse
+import datetime
+from pathlib import Path
+from abc import ABC, abstractmethod
+
+from pylele_api import ShapeAPI, Shape
+from pylele_config import Fidelity, Implementation, CARBON, GRAY, LITE_GRAY,DARK_GRAY, ORANGE, WHITE
+
+COLORS = {
+    'carbon': CARBON, 
+    'gray': GRAY, 
+    'lite_gray' : LITE_GRAY,
+    'dark_gray' : DARK_GRAY, 
+    'orange' : ORANGE, 
+    'white': WHITE
+}
+
+def make_or_exist_path(out_path):
+    """ Check a directory exist, and generate if not """
+
+    if not os.path.isdir(out_path):
+        # Path.mkdir(out_path)
+        os.makedirs(out_path)
+
+    assert os.path.isdir(out_path), f"Cannot export to non directory: {out_path}"
+
+def lele_solid_parser(parser=None):
+    """
+    LeleSolid Command Line Interface
+    """
+    if parser is None:
+        parser = argparse.ArgumentParser(description='LeleSolid Configuration')
+
+    ## options ######################################################
+    parser.add_argument("-i", "--implementation", help="Underlying engine implementation, default cadquery",
+                        type=Implementation, choices=list(Implementation), default=Implementation.BLENDER)
+    parser.add_argument("-f", "--fidelity", help="Mesh fidelity for smoothness, default low",
+                        type=Fidelity, choices=list(Fidelity), default=Fidelity.LOW)
+    parser.add_argument("-color", "--color", help="This solid is a cutting.", type=str,
```

Comment:
-c instead of -color?

Reply:
Sorry about the comment error.
Ehm I do not know if color is really needed... It was there in LelePart and that is why it is there. Should remove it? I do not know exactly what color is used for in pylele.